python/cudf/cudf/core/series.py

```diff
@@ -538,7 +544,13 @@ def __init__(
                 data = {}
 
         if not isinstance(data, ColumnBase):
-            data = column.as_column(data, nan_as_null=nan_as_null, dtype=dtype)
+            data = column.as_column(
+                data,
+                nan_as_null=nan_as_null,
+                dtype=dtype,
+                copy=copy,
+                fastpath=True,
+            )
```

Comment:
Isn't this what we really ought to do if `copy=True`?

```suggestion
            data = column.as_column(
                data,
                nan_as_null=nan_as_null,
                dtype=dtype,
            )
            if copy:
                data = data.copy(deep=True)
```

I realize there's a redundant copy we can avoid here if `data` is originally a host object (let's put that aside for a moment).

Reply:
Yup, made the change to this simpler version with help of `as_buffer` that accomplishes marking the buffer as zero-copied.